### PR TITLE
updated based on crdcdh-260

### PR DESCRIPTION
**Pull request by vshand11**
updated codes for file and metadata validation and log file uploading for crdcdh-260

### CODE REVIEW

File: copier.py [hunk 1]
```diff
@@ -152,8 +152,20 @@ def copy_file(self, file_info, overwrite, dryrun, verify_md5=False):
                        }
 
             if dryrun:
+                if verify_md5:  #assume validate file if verify_md5 is true and dryrun is true
+                    if _is_local(org_url): #validate org_file_size against real local file size
```
Comment (n2iw) [inline copier.py:156]: This size validation should always be executed, no matter dryrun or verfify_md5 have been set or not.

**Review by n2iw — CHANGES_REQUESTED** [review]

File: file_copier.py [hunk 2]
```diff
@@ -6,9 +6,10 @@
 from collections import deque
 
 from bento.common.sqs import Queue, VisibilityExtender
-from bento.common.utils import get_logger, get_uuid, LOG_PREFIX, UUID, get_time_stamp, removeTrailingSlash, load_plugin
+from bento.common.utils import get_logger, get_log_file, get_uuid, LOG_PREFIX, UUID, get_time_stamp, removeTrailingSlash, load_plugin
 from copier import Copier
 from file_copier_config import MASTER_MODE, SLAVE_MODE, SOLO_MODE, Config
+from bento.common.s3 import upload_log_file
```
Comment (n2iw) [inline file_copier.py:12]: To make this import work, we need to update the submodule too.

**Review by n2iw — CHANGES_REQUESTED** [review]

File: file_copier.py [hunk 3]
```diff
@@ -294,6 +297,16 @@ def copy_all(self):
...
 
+        #upload log file into configured upload_log_dir
+        ori_log_file = get_log_file()
+        if self.upload_log_dir:
+            try:
+                upload_log_file(self.upload_log_dir, ori_log_file)
+                self.log.error(f'Uploading log file {ori_log_file} succeeded!')
+            except Exception as e:
+                self.log.debug(e)
+                self.log.error(f'Uploading log file {ori_log_file} failed!')
```
[the remark this answers is not in the document]
Comment (vshand11) [reply file_copier.py:308]: Updated.  Thanks!

File: loader.py [hunk 4]
```diff
@@ -166,12 +166,12 @@ def process_arguments(args, log):
 
     return config
 
-
-def upload_log_file(bucket_name, folder, file_path):
-    base_name = os.path.basename(file_path)
-    s3 = S3Bucket(bucket_name)
-    key = f'{folder}/{base_name}'
-    return s3.upload_file(key, file_path)
+# use util function imported
```
[the remark this answers is not in the document]
Comment (vshand11) [reply loader.py]: Updated.

File: loader.py [hunk 5]
```diff
@@ -255,13 +256,18 @@ def main():
...
-    if config.s3_bucket and config.s3_folder:
-        result = upload_log_file(config.s3_bucket, f'{config.s3_folder}/logs', log_file)
-        if result:
-            log.info(f'Uploading log file {log_file} succeeded!')
-        else:
-            log.error(f'Uploading log file {log_file} failed!')
-
+    log_file = get_log_file()
+    #set default log file upload dir
+    dest_log_dir = f's3://{config.s3_bucket}/{config.s3_folder}/logs'
```
[the remark this answers is not in the document]
Comment (vshand11) [reply loader.py]: Updated logic to handle the s3 bucket/folder are not set.

File: copier.py (hunk 1 above)
Comment (vshand11) [reply copier.py:156]: Updated, thanks!

File: loader.py [hunk 6]
```diff
@@ -235,7 +235,8 @@ def main():
                 log.info(restore_cmd)
             if load_result == False:
```
[the remark this answers is not in the document]
Comment (vshand11) [reply loader.py:227]: Updated, thanks!

**Review by n2iw — APPROVED** [review]

File: loader.py [hunk 7]
```diff
@@ -255,13 +245,26 @@ def main():
         if restore_cmd:
             log.info(restore_cmd)
 
-    if config.s3_bucket and config.s3_folder:
-        result = upload_log_file(config.s3_bucket, f'{config.s3_folder}/logs', log_file)
-        if result:
+    log_file = get_log_file()
+    dest_log_dir = ''
```
[the remark this answers is not in the document]
Comment (vshand11) [reply loader.py]: Updated, thanks!

**Pull request merged**